src/project/mod.rs

```diff
@@ -127,15 +130,30 @@ impl ProjectController {
                 );
             }
             ProjectUpdate::CairoProjectToml(maybe_project_config) => {
+                if let Some(project_config) = maybe_project_config {
```

Comment:
Question: Why do we only add this logic on the `CairoProjectToml` update, when corelib has both this and `Scarb.toml` ? Is it guaranteed to have precedence?